### DISCUSSION TITLE
Wayland vs X11

### DISCUSSION THREAD

**Discussion by Thorium9O** [AlexandreRouma/SDRPlusPlus#721] [General]
Is there a way to change the sdrpp from the Wayland package to the X11 version?
Thanks in advance,

**Comment by the-gobleo-tinkerer** [comment]
On my Wayland system, SDR++ runs under XWayland, so it's already using (emulated) X11. I built SDR++ from source, using commit 6f747da.

Please could you clarify what you are trying to do?

**Comment by Thorium9O** [comment]
When I first install srdpp on Manjaro there was a Wayland and x11 option. I've tried both and the x11 option seems to work better. I looking to switch my current installation to the other option. 
Thanks for the help,

**Reply by the-gobleo-tinkerer** [reply]
Since you are on Manjaro I assume you installed SDR++ from the AUR. I tried both `sdrpp` and `sdrpp-git`, as well as building from source, but none of them gives an option between X and Wayland, so I don't know where you got that option from. Could you describe in detail how you installed SDR++, and where you saw that option?

**Comment by Thorium9O** [comment]
I got it from the AUR link on the git hub page. On the first install using makepkg -si but not any of the reinstalls it gives a "Enter a selection (default=1): " option with one Wayland and one X11.

**Reply by the-gobleo-tinkerer** [reply]
Ah I see. This selection is not provided by SDR++ itself, but by pacman. One of sdrpp's dependencies, `glfw`, has several versions (`glfw-wayland`, `glfw-x11`, among others), so pacman asks you to choose which `glfw` you want.

To see which one you have installed, do `pacman -Qs glfw`. To switch to the X11 GLFW, do `sudo pacman -S glfw-x11`, and confirm replacing `glfw-wayland` when asked.

By the way - if your problem with the Wayland version is the window missing the title bar (no close button etc.), you could try `glfw-wayland-minecraft-libdecoration` instead, which adds those controls (although it doesn't follow your system theme).

**Comment by Thorium9O** [comment]
Yep, that fixes it. Thanks for the help.
